pepper-apis/src/main/java/org/broadinstitute/ddp/log/SlackAppender.java

```diff
@@ -115,49 +149,71 @@ private String getStringifiedStackTrace(ILoggingEvent e) {
-            String stackTrace = getStringifiedStackTrace(e);
-
-            String message = MESSAGE.replace(TITLE, exceptionMessage);
-            message = message.replace(STACK_TRACE, stackTrace);
-
-            List<Attachment> attachments = new ArrayList<>();
-            String hostName = HostUtil.getHostName();
-
-            attachments.add(Attachment.builder().text("from " + hostName).build());
-            String logBreadCrumb = null;
-            if (e.getMDCPropertyMap().containsKey(MDCLogBreadCrumbFilter.LOG_BREADCRUMB)) {
-                logBreadCrumb = e.getMDCPropertyMap().get(MDCLogBreadCrumbFilter.LOG_BREADCRUMB);
-            }
-            if (StringUtils.isNotBlank(logBreadCrumb)) {
-                attachments.add(Attachment.builder().text("grep for `" + logBreadCrumb + "`").build());
+            synchronized (messagesToSend) {
+                if (!isQueueFull()) {
+                    String exceptionMessage = getExceptionMessage(e);
+                    String stackTrace = getStringifiedStackTrace(e);
+
+                    String message = MESSAGE.replace(TITLE, exceptionMessage);
+                    message = message.replace(STACK_TRACE, stackTrace);
+
+                    SlackMessagePayload messagePayload = new SlackMessagePayload(message, channel, "Pepper",
+                            ":nerd_face:");
+                    messagesToSend.add(messagePayload);
+                }
             }
+        }
+    }
 
-            Payload messagePayload = Payload.builder()
-                    .channel(channel)
-                    .text(message)
-                    .iconEmoji(":nerd_face:")
-                    .attachments(attachments)
-                    .username("Peppa").build();
-            sendMessage(messagePayload);
+    private void sendQueuedMessages() {
+        LOG.info("Sending {} messages to slack", messagesToSend.size());
+        synchronized (messagesToSend) {
+            for (SlackMessagePayload payload : messagesToSend) {
+                sendMessage(payload);
+            }
+            messagesToSend.clear();
         }
+        notifyAll();
     }
 
-    private void sendMessage(Payload payload) {
-        WebhookResponse response = null;
+    private void sendMessage(SlackMessagePayload payload) {
+
+        HttpRequest httpRequest = HttpRequest.newBuilder()
+                    .uri(slackHookUrl)
+                    .POST(HttpRequest.BodyPublishers.ofString(new Gson().toJson(payload)))
+                    .header(HttpHeaders.CONTENT_TYPE, ContentType.APPLICATION_JSON.getMimeType())
+                    .build();
         try {
-            response = slack.send(slackHookUrl, payload);
-            if (response.getCode() != 200) {
-                throw new IOException("Could not post " + payload + " to slack.  Hook returned " + response
-                        .getMessage());
+            var httpClient = HttpClient.newHttpClient();
```

Comment:
@MocanaAtBroad taking your advice here and just dropping down to the API directly...

Reply:
I did attempt a different approach earlier that involved closing up 3 different potential resources in `okhttp`, wrapped up in `jslack`.  This is much simpler.

Reply:
Oh! My suggestion was switching to Slack's own official Java library instead the third-party library given how many difficulties it was creating.
But looks like you managed to do it using REST, so awesome! Less stuff to worry about!